pkg/ipcache/metadata_test.go

```diff
@@ -413,9 +413,9 @@ func TestInjectWithLegacyAPIOverlap(t *testing.T) {
 	identityReferences++
 	for i := 0; i < 2; i++ {
 		IPIdentityCache.UpsertLabels(prefix, labels, source.CustomResource, resource)
-		// Need to wait for the label injector to finish; easiest just to remove it
-		IPIdentityCache.controllers.RemoveControllerAndWait(LabelInjectorName)
 	}
+	// Need to wait for the label injector to finish; easiest just to remove it
+	IPIdentityCache.ShutdownLabelInjection()
```

Comment:
Was this happening inside the loop for a reason? :thinking: 

Given that this particular test is about testing reference counts and this has an arbitrary `2` for inserting the exact same labels, I think this is trying to ensure that the inner function completes twice. However if this line is moved outside the loop then we don't know if the inner `InjectLabels()` runs twice.

Conclusion: This *must* ensure that the inner loop in `InjectLabels()runs twice. This change as-is breaks the test.